### PR TITLE
feat: Added ResNet-18/50 models and training pipeline

### PR DESCRIPTION
**Pull request by mikdzw**
- Implemented ResNet-18 and ResNet-50 architectures
- Added a completed training script with validation and checkpointing
- Achieved 97.47% validation accuracy on OrganAMNIST using ResNet-18 (https://huggingface.co/mikidzw/organamnist-resnet18/blob/main/resnet18_best.pth)
- Added dependencies: tqdm, huggingface-hub to .toml

### CODE REVIEW

**Comment by InfiniteLobster** [comment]
Hi, I have quick (potentially stupid) question. Why implementing models from scratch and not using ready version from torchvision (with adaptation to our dataset images(although I'm not sure if we decided if we are going to use 28x28, 64x64, 128x128 or 224x224))?
